test/io/event/selector/file_io.rb

```diff
@@ -29,32 +29,42 @@
 		let(:file) {Tempfile.new}
 		
 		it "can read using a buffer" do
+			write_result = nil
+			read_result = nil
+			
 			writer = Fiber.new do
 				buffer = IO::Buffer.new(128)
 				file.seek(0)
-				expect(selector.io_write(Fiber.current, file, buffer, 128)).to be == 128
+				write_result = selector.io_write(Fiber.current, file, buffer, 128)
 			end
 			
 			reader = Fiber.new do
 				buffer = IO::Buffer.new(64)
 				file.seek(0)
-				expect(selector.io_read(Fiber.current, file, buffer, 1)).to be == 64
+				read_result = selector.io_read(Fiber.current, file, buffer, 1)
 			end
 			
 			writer.transfer
 			reader.transfer
+			
+			selector.select(0)
+
+			expect(write_result).to be == 128
+			expect(read_result).to be == 64
```

Comment:
So, I'm trying to make the above test pass for uring. I added a line with `selector.select(0)` and I'm now getting a weird error. I also enabled `DEBUG` and `DEBUG_IO_READ` flags, the full output is below.

I spent considerable time debugging this, but it was a rabbit hole... the error I'm getting for this test is:

- `NoMethodError: undefined method 'alive?' for 2.0:Float`.  Weird, we're not using 2.0 anywhere.
- The error backtrace isn't showing this, but the error happens when transferring to a fiber and checking if it's `alive?` on this line:

https://github.com/socketry/io-event/blob/3b43389ef480e87b60c540a987473864ba5093d5/ext/io/event/selector/uring.c#L617

- So, how are we resuming a `2.0` Float? I think this is related to this debug output line: `cqe res=0 user_data=0x2`. Note that `user_data` is `0x2`, when it should be the address of the fiber. I investigated this, but couldn't figure it out. Any pointers or ideas are appreciated, thanks.

Full output:

```
vagrant@ubuntu-impish:/vagrant/io-event$ bundle exec sus test/io/event/selector/file_io.rb
io_write:io_uring_prep_write(fiber=0x7fd3d8ed2278)                                                    
io_read(6, +0, 64)                                                                                    
io_read:io_uring_prep_read(fiber=0x7fd3d8ed21b0)                                                      
cqe res=0 user_data=0x2                                                                               
0 assertions                                                                                          
                                                                                                      
describe IO::Event::Selector::URing it behaves like file io with a file it can read using a buffer test/io/event/selector/file_io.rb:105:31
✗ Unhandled exception NoMethodError: undefined method `alive?' for 2.0:Float       
test/io/event/selector/file_io.rb:50:in `select'                                                      
test/io/event/selector/file_io.rb:50:in `block (4 levels) in <top (required)>'                   
/usr/local/lib/ruby/gems/3.1.0/gems/sus-0.6.2/lib/sus/it.rb:33:in `block (2 levels) in call'
/usr/local/lib/ruby/gems/3.1.0/gems/sus-0.6.2/lib/sus/base.rb:19:in `around'
```